### PR TITLE
feat: i18n with angular localize

### PR DESCRIPTION
**Pull request by klopfdreh**
**This PR provides a complete i18n integration for spring-cloud-dataflow-ui.** (at least I hope I caught every translation required occurrence 😁 )

I also provide a complete translation into german language with this PR.

Screenshot:
Bildschirmfoto 2021-12-07 um 12 02 00

Some important commands:

`ng extract-i18n --format=xlf2` - extract all translations into a messages.xlf which should be renamed into messages.en.xlf and moved into `src/locale/`. All other language files should be compared against this one.
`ng serve --open -c=de` - runs the ui with a given configuration in this case de (German)

I also adjusted some of the script commands.

This PR addresses: https://github.com/spring-cloud/spring-cloud-dataflow-ui/issues/701

### CODE REVIEW

**Comment by klopfdreh** [comment]
If the PR https://github.com/spring-cloud/spring-cloud-dataflow-ui/pull/1786 is merged first this one would require an update of localize to "13.0.1"

**Comment by klopfdreh** [comment]
I adjusted the command to extract i18n a bit:
`npm run i18n:extract` - to extract the translations and place them directly into src/locale/messages.en.xlf

**Comment by klopfdreh** [comment]
Force-Push on my Repository? 😨 Anyway feel free I didn't restrict it. 😄

**Comment by oodamien** [comment]
Hi @klopfdreh,
Sorry for that, we are working on the PR:
- Update Angular
- Dynamic language switch

**Comment by klopfdreh** [comment]
Great to see! 😃 - if I can help further more just ping me!

File: ui/angular.json [hunk 1]
```diff
@@ -9,10 +9,15 @@
       "projectType": "application",
       "prefix": "app",
       "i18n": {
-        "sourceLocale": "en-US",
+        "sourceLocale": {
+          "code": "en",
```
Comment (klopfdreh) [inline ui/angular.json:13]: Nice to see that you can apply "en" for source locale - I just tried that, but without a JSON object and it failed. Good to see that it works, now. 👍

**Comment by oodamien** [comment]
Close in favor of #1794 (easier maintenance, dynamic translation)

**Pull request closed without merging**